frontend/src/components/ui/SaveAsPipelineButton.jsx

```diff
@@ -29,7 +30,7 @@ export default function SaveAsPipelineButton() {
     pipelineSpecs["sink"] = pipeline.path;
     pipelineSpecs["build"] = pipeline.path;
     pipelineSpecs["name"] = pipeline.name;
-    pipelineSpecs["id"] = pipeline.id;
+    pipelineSpecs["id"] = generateId("pipeline");
```

Comment:
Is this for a bug fix? If we're changing it here, then should we also change `SavePipelineButton`? The behaviour of `SavePipelineButton` when the pipeline hasn't been saved is the same as `SaveAsPipelineButton`.

Reply:
It is a bug fix--what ends up happening if you save two pipelines with the same ID they will not be able to load in the UI (they will overwrite each other).

With Save the assumption is that it doesn't exist yet on disk, or if it does we already know the path. In save we could check if there is a local path, and if not we could generate a new id to be safe.